src/Extension/KernelExtensions.cs

```diff
@@ -32,18 +34,62 @@ public static class KernelExtensions
 
         public static T UseProgressiveLearning<T>(this T kernel) where T : Kernel, IKernelCommandHandler<ParseNotebook>
         {
-            var argument = new Argument<FileInfo>("file");
-            
+            Option<Uri> fromUrlOption = new Option<Uri>(
+                "--from-url",
+                "Specify lesson source URL" );
+
+            Option<FileInfo> fromFileOption = new Option<FileInfo>(
+                "--from-file",
+                description: "Specify lesson source file",
+                parseArgument: result =>
+                {
+                    var filePath = result.Tokens.Single().Value;
+                    var fromUrlResult = result.FindResultFor(fromUrlOption);
+
+                    if (fromUrlResult is not null)
+                    {
+                        result.ErrorMessage = $"The {fromUrlResult.Token.Value} and {(result.Parent as OptionResult).Token.Value} cannot be used together";
```

Comment:
should be the ... and ... options cannot be used together.
Might be good to add a test for this